tensorboard/examples/plugins/example_raw_scalars/tensorboard_plugin_example_raw_scalars/static/components.js

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+
+import * as Model from './model.js';
+
+/**
+ * @param {string} run
+ * @return {!Promise<!DocumentFragment>}
+ */
+export async function createPreviews(run) {
+  const fragment = document.createDocumentFragment();
+
+  const tags = await Model.getTags(run);
+  if (!tags || !tags.length) {
+    const messageElement = createElement('h2');
+    messageElement.textContent = 'No tags found.';
+    fragment.appendChild(messageElement);
+    return fragment;
+  }
+
+  /**
+   * For each tag, build UI in this form:
+   *   <div class="preview">
+   *     <div class="tagname">${tag}</div>
+   *     <textarea class="preview-text">${result}</textarea>
+   *   </div>
+   */
+  for (let tag of tags) {
+    const scalars = await Model.getScalars(run, tag);
+    const previewEl = createElement('div', 'preview');
+    const tagNameEl = createElement('div', 'tagname');
+    const textPreviewEl = createElement('textarea', 'preview-text');
+    tagNameEl.textContent = tag;
+    textPreviewEl.textContent =
+        scalars ? JSON.stringify(scalars) : 'No scalar data found.';
+    previewEl.appendChild(tagNameEl);
+    previewEl.appendChild(textPreviewEl);
+    fragment.appendChild(previewEl);
+  }
+  return fragment;
+}
+
+/**
+ * @param {!Array<string>} runs
+ * @param {function()} onchange
+ * @return {!HTMLSelectElement}
+ */
+export async function createRunSelector(onchange) {
```

Comment:
I would expect to disentangle data fetching from the DOM creation. WDYT?

Reply:
Done.